smqtk_classifier/impls/classify_descriptor_supervised/sklearn_svm.py

```diff
@@ -0,0 +1,449 @@
+
+from smqtk_classifier.interfaces.classify_descriptor import CLASSIFICATION_DICT_T
+from smqtk_classifier.interfaces.classify_descriptor_supervised import ClassifyDescriptorSupervised
+
+
+LOG = logging.getLogger(__name__)
+SVM_PARAM_MAPPING_T = Mapping[str, Union[int, float, str]]
+
+
+try:
+    # noinspection PyPackageRequirements
+    import scipy.stats  # type: ignore
+except ImportError:
+    warnings.warn(
+        "scipy.stats not importable: SkLearnSvmClassifier will not be usable."
+    )
+    scipy = None
+
+try:
+    from sklearn import svm
+except ImportError:
+    warnings.warn(
+        "svm not importable: SkLearnSvmClassifier will not be usable."
+    )
+    svm = None
+
+
+class SkLearnSvmClassifier (ClassifyDescriptorSupervised):
+    """
+    Classifier that uses SkLearn for support-vector machine functionality.
+
+    Model file paths are optional. If they are given and the file(s) exist,
+    we will load them. If they do not, we treat the path(s) as the output
+    path(s) for saving a model after calling ``train``. If this is None
+    (default), no model is loaded nor output via training, thus any model
+    trained will only exist in memory during the lifetime of this instance.
+
+    **Note**: *If pickled without a having model file paths configured, this
+    implementation will write out temporary files upon pickling and loading.
+    This is required because the model instance is not transportable via
+    serialization due to libSVM being an external C library.*
+
+    :param svm_model_uri: Path to the libSVM model file.
+    :param svm_label_map_uri: Path to the pickle file containing this
+        model's output labels.
+    :param train_params: SVM parameters used for training. See libSVM
+        documentation for parameter flags and values. A parameter with no value
+        on the SVM CLI should be given an empty string value,
+        E.g. `{"-q": ""}`.
+    :param normalize: Normalize input vectors to training and
+        classification methods using ``numpy.linalg.norm``. This may either
+        be  ``None``, disabling normalization, or any valid value that
+        could be passed to the ``ord`` parameter in ``numpy.linalg.norm``
+        for 1D arrays. This is ``None`` by default (no normalization).
+    :param n_jobs:
+        Number of processes to use to parallelize prediction. If None or a
+        negative value, all cores are used.
+    """
+
+    # noinspection PyDefaultArgument
+    def __init__(
+        self,
+        svm_model_uri: Optional[str] = None,
+        svm_label_map_uri: Optional[str] = None,
+        train_params: SVM_PARAM_MAPPING_T = {
+            '-s': 0,  # C-SVC, assumed default if not provided
+            '-t': 0,  # linear kernel
+            '-b': 1,  # enable probability estimates
+            '-c': 2,  # SVM parameter C
+        },
+        normalize: Optional[Union[int, float, str]] = None,
+        n_jobs: Optional[int] = 4,
+    ):
```

Comment:
We can treat this as a "new" implementation: a.k.a. we don't need to mirror the libSVM's constructor etc. so exactly here. Let's keep it relevant to what *this* implementation wants, e.g. `train_params` was for libSVM, we can be more direct here for the scikit-learn requirements. More e.g. we could stick to this impl wrapping the `svm.SVC` algorithm for now; I don't think the other SVM variants were ever used.